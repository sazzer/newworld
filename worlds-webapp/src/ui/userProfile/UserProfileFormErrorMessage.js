// @flow

import React from 'react';
import {Message} from 'semantic-ui-react';
import {Interpolate} from 'react-i18next';

/** The message keys for the errors we can support */
const problemKeys = {
    'tag:grahamcox.co.uk,2018,users/problems/duplicate-username': 'userProfile.form.messages.error.duplicateUsername',
    'undefined': 'userProfile.form.messages.error.unexpected'
}

/** The flow type representing the props for the User Profile Form message */
type UserProfileFormErrorMessageProps = {
    errorCode?: string
};

/**
 * The message to display for the user profile save result
 */
export default function UserProfileFormErrorMessage(props: UserProfileFormErrorMessageProps) {
    let problemCode = props.errorCode || 'undefined';

    if (!problemKeys.hasOwnProperty(problemCode)) {
        problemCode = 'undefined';
    }
    const errorKey = problemKeys[problemCode];

    return (
        <Message error>
            <Interpolate i18nKey={errorKey} />
        </Message>
    )
}
