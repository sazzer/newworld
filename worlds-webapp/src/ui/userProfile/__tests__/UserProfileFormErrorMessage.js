// @flow

import React from 'react';
import {shallow} from 'enzyme';
import UserProfileFormErrorMessage from '../UserProfileFormErrorMessage';

/** Create the component to test */
function setup({errorCode}: {errorCode?: string} = {}) {
    return {
        element: shallow(<UserProfileFormErrorMessage errorCode={errorCode} />)
    };
}

describe('UserProfileFormErrorMessage', () => {
    it('Renders correctly with no error code', () => {
        const { element } = setup();

        expect(element).toMatchSnapshot();
    });
    it('Renders correctly with an unrecognized error code', () => {
        const { element } = setup({errorCode: 'unrecognized'});

        expect(element).toMatchSnapshot();
    });
    it('Renders correctly with a supported error code', () => {
        const { element } = setup({errorCode: 'tag:grahamcox.co.uk,2018,users/problems/duplicate-username'});

        expect(element).toMatchSnapshot();
    });
});
