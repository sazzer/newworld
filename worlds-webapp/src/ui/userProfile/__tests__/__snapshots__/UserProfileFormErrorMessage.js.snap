// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileFormErrorMessage Renders correctly with a supported error code 1`] = `
<Message
  error={true}
>
  <Interpolate
    className=""
    i18nKey="userProfile.form.messages.error.duplicateUsername"
  />
</Message>
`;

exports[`UserProfileFormErrorMessage Renders correctly with an unrecognized error code 1`] = `
<Message
  error={true}
>
  <Interpolate
    className=""
    i18nKey="userProfile.form.messages.error.unexpected"
  />
</Message>
`;

exports[`UserProfileFormErrorMessage Renders correctly with no error code 1`] = `
<Message
  error={true}
>
  <Interpolate
    className=""
    i18nKey="userProfile.form.messages.error.unexpected"
  />
</Message>
`;
